src/group/libp2p_group_gossip_server.erl

```diff
@@ -298,27 +298,39 @@ connections(Kind, #state{workers=Workers}) ->
                         Acc
                 end, [], Workers).
 
-assign_target(WorkerPid, TargetAddrs, State=#state{workers=Workers}) ->
+assign_target(WorkerPid, WorkerRef, TargetAddrs, State=#state{workers=Workers}) ->
     case length(TargetAddrs) of
         0 ->
-            case lookup_worker(WorkerPid, #worker.pid, State) of
-                #worker{kind=seed, target=SelectedAddr} when SelectedAddr /= undefined ->
+            %% the ref is stable across restarts, so use that as the lookup key
+            case lookup_worker(WorkerRef, #worker.ref, State) of
+                Worker=#worker{kind=seed, target=SelectedAddr, pid=P} when SelectedAddr /= undefined ->
                     %% don't give up on the seed nodes in case we're entirely offline
                     %% we need at least one connection to bootstrap the swarm
                     ClientSpec = {?GROUP_PATH, {libp2p_gossip_stream, [?MODULE, self()]}},
-                    libp2p_group_worker:assign_target(WorkerPid, {SelectedAddr, ClientSpec});
+                    libp2p_group_worker:assign_target(WorkerPid, {SelectedAddr, ClientSpec}),
+                    %% check if this worker got restarted
+                    case WorkerPid /= P of
```

Comment:
mind calling `P` StoredWorkerPid or something to make it clearer?

Reply:
Fixed